dbms/src/Flash/ResourceControl/LocalAdmissionController.h

```diff
@@ -128,26 +151,190 @@ class ResourceGroup final : private boost::noncopyable
+        acquire_num = (acquire_num > 0.0 ? acquire_num : 0.0);
+
+        LOG_TRACE(
+            log,
+            "acquire num for rg {}: avg_speed: {}, remaining_ru: {}, base: {}, amplification: {}, "
+            "acquire num: {}",
+            name,
+            avg_speed,
+            remaining_ru,
+            base,
+            amplification,
+            acquire_num);
+        return acquire_num;
+    }
+
+    void updateNormalMode(double add_tokens, double new_capacity)
+    {
+        assert(add_tokens >= 0);
+
+        std::lock_guard lock(mu);
+        bucket_mode = TokenBucketMode::normal_mode;
+        if (new_capacity <= 0.0)
+        {
+            burstable = true;
+            return;
+        }
+        auto config = bucket->getConfig();
+        std::string ori_bucket_info = bucket->toString();
+
+        config.tokens += add_tokens;
+        config.fill_rate = 0;
+        config.capacity = new_capacity;
+        bucket->reConfig(config);
+        LOG_INFO(
+            log,
+            "token bucket of rg {} reconfig to normal mode. from: {}, to: {}",
+            name,
+            ori_bucket_info,
+            bucket->toString());
+    }
+
+    void updateTrickleMode(double add_tokens, double new_capacity, int64_t trickle_ms)
+    {
+        assert(add_tokens > 0.0);
+        assert(trickle_ms > 0);
+
+        std::lock_guard lock(mu);
+        if (new_capacity <= 0.0)
+        {
+            burstable = true;
+            return;
+        }
+
+        bucket_mode = TokenBucketMode::trickle_mode;
+
+        const double trickle_sec = static_cast<double>(trickle_ms) / 1000;
+        const double new_fill_rate = add_tokens / trickle_sec;
+        RUNTIME_CHECK_MSG(
+            new_fill_rate > 0.0,
+            "token bucket of {} reconfig to trickle mode failed. add_tokens: {} trickle_ms: {}, trickle_sec: {}",
+            name,
+            add_tokens,
+            trickle_ms,
+            trickle_sec);
+
+        std::string ori_bucket_info = bucket->toString();
+        bucket->reConfig(TokenBucket::TokenBucketConfig(bucket->peek(), new_fill_rate, new_capacity));
+        stop_trickle_timepoint = std::chrono::steady_clock::now() + std::chrono::milliseconds(trickle_ms);
+        LOG_INFO(
+            log,
+            "token bucket of rg {} reconfig to trickle mode: from: {}, to: {}",
+            name,
+            ori_bucket_info,
+            bucket->toString());
+    }
+
+    // If we have network problem with GAC, enter degrade mode.
+    void toDegrademode()
+    {
+        std::lock_guard lock(mu);
+        if (burstable || bucket_mode == degrade_mode)
+            return;
+
+        bucket_mode = TokenBucketMode::degrade_mode;
+        double avg_speed = bucket->getAvgSpeedPerSec();
+        auto config = bucket->getConfig();
+        std::string ori_bucket_info = bucket->toString();
+
+        config.fill_rate = avg_speed;
+        bucket->reConfig(config);
+        LOG_INFO(
```

Comment:
There are too many info logs here, and too many info logs will cause the log of tidb cloud to be lost.

Reply:
The frequency of calling `toDegradeMode()` is not high, it only occurs when there are network issues. I think it ok to do `LOG_INFO` for `toDegradeMode()`.

And I have changed most `LOG_INFO` to `LOG_DEBUG` for LAC.